### ISSUE TITLE
non-zero baseline

### ISSUE THREAD

**Issue by ali-amirahmadii** [sherlcok314159/Integrated-Gradients#1]
<p>
I would like to add a non-zero baseline for calculating integrated gradients using your codebase. My goal is to use a custom baseline to improve interpretability. Below is the code I’ve adapted to introduce a custom baseline—could you confirm if this approach is correct?
</p>

<ol>
    <li><strong>Set the Custom Baseline</strong>: I defined a baseline using a masked text input and extracted embeddings as follows:
        <pre><code>
baseline_text = ["I am [MASK] because the weather is extremely good!"]
baseline_inputs = tk(baseline_text, max_length=128, return_tensors="pt", truncation=True, padding=True)

with torch.no_grad():  # No gradients needed here
    baseline_embed = model.bert.embeddings(
        input_ids=baseline_inputs["input_ids"],
        token_type_ids=baseline_inputs["token_type_ids"]
    )
        </code></pre>
    </li>

  

Integrated Gradients Function with Baseline: Here’s the modified function, which now accepts a baseline and uses it for interpolation with the original input embeddings.z
        <pre><code>
```
from torch.autograd import grad
from copy import deepcopy

def get_integrated_gradients(
  model,
    input_ids,
    token_type_ids,
    attention_mask,
    baseline=None,
    num_steps=4
):
    # Get the word embedding matrix for the original input
    input_embed = model.bert.embeddings(input_ids=input_ids, token_type_ids=token_type_ids)
    backup_embed = deepcopy(input_embed.data)

    # Use the provided baseline embedding if available; otherwise, use zeros
    if baseline is None:
        baseline = torch.zeros_like(backup_embed)

    grads = []
    for step in range(num_steps + 1):
        alpha = step / num_steps
        # Interpolate between the baseline and the input embeddings
        input_embed.data = baseline + alpha * (backup_embed - baseline)

        # Forward pass to get model outputs
        bert_outputs = model(input_ids, attention_mask, token_type_ids)
        logits, hidden_states = bert_outputs.logits, bert_outputs.hidden_states
        logits_l = logits.max(dim=1).values  # Max reduction on logits for label output

        # Calculate the gradients of the output with respect to the embedding layer output
        embed_out = hidden_states[0]
        g = grad(logits_l, embed_out, grad_outputs=torch.ones_like(logits_l))[0]
        grads.append(g)

    labels = logits.argmax(dim=1).tolist()
    grads = torch.stack(grads)  # Stack gradients along steps
    grads = (grads[:-1] + grads[1:]) / 2  # Average adjacent steps
    avg_grads = grads.mean(dim=0)

    integrated_grads = embed_out * avg_grads
    return integrated_grads, labels
```
        
# Call the integrated gradients function with the custom baseline
grads, labels = get_integrated_gradients(model, baseline=baseline_embed, **inputs)
        </code></pre>
    </li>
</ol>

<p>
Currently, the code runs without errors, but I would like confirmation on whether this implementation effectively integrates the baseline in the way intended.
</p>
